app/src/main/java/ai/brokk/analyzer/TreeSitterAnalyzer.java

```diff
@@ -3601,28 +3680,39 @@ protected AnalyzerState runTypeAnalysis(AnalyzerState baseState) {
 
         Map<CodeUnit, CodeUnitProperties> updatedCodeUnitState = new ConcurrentHashMap<>(baseState.codeUnitState());
 
+        // Count total classes to process
+        int totalClasses = (int) baseState.codeUnitState().keySet().stream()
+                .filter(CodeUnit::isClass)
+                .count();
+        var progressReporter = new DebouncedProgressReporter(totalClasses, "Computing type hierarchies", 100);
+
         int parallelism = Runtime.getRuntime().availableProcessors();
         try (var fjp = new java.util.concurrent.ForkJoinPool(parallelism)) {
-            fjp.execute(() -> {
-                baseState.codeUnitState().entrySet().parallelStream()
-                        .filter(e -> e.getKey().isClass())
-                        .forEach(entry -> {
-                            CodeUnit cu = entry.getKey();
-                            CodeUnitProperties props = entry.getValue();
-                            List<CodeUnit> supers = delegateForTypes.computeSupertypes(cu);
-                            if (!Objects.equals(props.supertypes(), supers)) {
-                                updatedCodeUnitState.put(
-                                        cu,
-                                        new CodeUnitProperties(
-                                                props.children(),
-                                                props.signatures(),
-                                                props.ranges(),
-                                                props.rawSupertypes(),
-                                                supers,
-                                                props.hasBody()));
-                            }
-                        });
-            });
+            fjp.submit(() -> {
```

Comment:
Could possibly make this simpler with `invoke` (which I believe is blocking) instead of `submit` + `join`.

Reply:
Nice thought.  I decided to take a look at the source of invoke(), and you're right, it looks like it is pretty much the same as submit+join.  The only thing is invoke requires that I pass a ForkJoinTask, which makes me have to wrap the lambda with ForkJoinTask.adapt() which makes it less clear that it's simpler.